tests/testthat/test-match_site_ids.R

```diff
@@ -0,0 +1,165 @@
+context("match_site_ids_by_location")
+test_that("match_site_ids_by_location returns full mapping when all IDs exist", {
+  target <- data.frame(
+    site_id = c("A", "B"),
+    lat = c(34.0, 35.0),
+    lon = c(-118.0, -119.0)
+  )
+  reference <- data.frame(
+    site_id = c("A", "B"),
+    lat = c(34.0, 35.0),
+    lon = c(-118.0, -119.0)
+  )
```

Comment:
Nit: If these are supposed to really be identical, copy one so readers don't have to play spot the difference
```suggestion
  reference <- target
```